pandas/core/groupby/generic.py

```diff
@@ -236,9 +236,11 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             kwargs = {}
 
         if isinstance(func, str):
-            if maybe_use_numba(engine):
+            if maybe_use_numba(engine) and engine is not None:
                 # Not all agg functions support numba, only propagate numba kwargs
-                # if user asks for numba
+                # if user asks for numba, and engine is not None
+                # (if engine is None, the called function will handle the case where
+                # numba is requested via the global option)
                 kwargs["engine"] = engine
                 kwargs["engine_kwargs"] = engine_kwargs
```

Comment:
If engine is None and engine_kwargs is not None, don't they end up getting ignored here?

Reply:
As long as `engine_kwargs` is not None, yes, that sounds right to me.